Mathlib/Tactic/Linter/AdmitLinter.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Adomas Baliuka
+-/
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+# The "admit" linter
+
+The "admit" linter flags usages of the `admit` tactic.
+
+The tactics `admit` and `sorry` are synonyms.
+The use of `sorry` is much more common and should be preferred.
+
+This linter is an incentive to discourage uses of `admit`, without being a ban.
+-/
+
+open Lean Elab
+
+namespace Mathlib.Linter.admit
+
+/-- The admit linter emits a warning on usages of `admit`. -/
+register_option linter.admit : Bool := {
+  defValue := true
```

Comment:
This setting enables the linter in *any* file which imports this (or a file importing this one). As you want ~all of mathlib to be linted, that implies projects *depending* on mathlib will also see this linter. The question to decide is whether this is what we want.
I tend towards "no": this is purely a decision about code style, which mathlib should not impose on other projects.
Thus, the ideal long-term fix is to *disable* this linter by default, but *enable* it in mathlib (and the archive and counterexamples).
That said: there's another PR open about doing exactly this - so you're very welcome to argue the other PR should modify this linter instead!

Reply:
Mathlib does not allow commits that contain `sorry` or `admit` anyway. So I'm not sure what the linter would be doing in that case?

Reply:
The linter would not affect code that gets merged into mathlib, but it would still highlight "in-development" `admit`s.  I also agree that the default value should be `false` anyway: the admit command is defined in core and it should really be the decision of the core developers if they want the `sorry`-synonym to continue existing or not.  The linter gives the opportunity to projects downstream of mathlib to choose whether or not they allow it.

Reply:
Set default to `false`.